tests/phpunit/tests/media.php

```diff
@@ -3617,6 +3619,69 @@ private function reset_omit_loading_attr_filter() {
 		// Clean up the above filter.
 		remove_filter( 'wp_omit_loading_attr_threshold', '__return_null', 100 );
 	}
+
+	/**
+	 * Test the wp_default_image_output_mapping function.
+	 *
+	 * @ticket 55443
+	 */
+	public function test_wp_default_image_output_mapping() {
+		$mapping = wp_default_image_output_mapping( array() );
+		$this->assertSame( array( 'image/jpeg' => 'image/webp' ), $mapping );
+	}
+
+	/**
+	 * Test that wp_default_image_output_mapping doesn't overwrite existing mappings.
+	 *
+	 * @ticket 55443
+	 */
+	public function test_wp_default_image_output_mapping_existing() {
+		$mapping = array( 'mime/png' => 'mime/webp' );
+		$mapping = wp_default_image_output_mapping( $mapping );
+		$this->assertSame(
+			array(
+				'mime/png'   => 'mime/webp',
+				'image/jpeg' => 'image/webp',
+			),
+			$mapping
+		);
+	}
+
+	/**
+	 * Test that the image editor default output for JPEGs is WebP.
+	 *
+	 * @ticket 55443
+	 */
+	public function test_wp_image_editor_default_output_maps_to_webp() {
+		remove_filter( 'image_editor_output_format', '__return_empty_array' );
+
+		$editor = wp_get_image_editor( DIR_TESTDATA . '/images/canola.jpg' );
+		if ( is_wp_error( $editor ) ) {
+			$this->markTestSkipped( $editor->get_error_message() );
+		}
+		$resized = $editor->resize( 100, 100, false );
+
+		if ( is_wp_error( $resized ) ) {
+			$this->markTestSkipped( $resized->get_error_message() );
+		}
+
+		$dest_file = $editor->generate_filename();
+		$saved     = $editor->save( $dest_file );
+
+		if ( is_wp_error( $saved ) ) {
+			$this->markTestSkipped( $saved->get_error_message() );
+		}
+
+		if ( $editor->supports_mime_type( 'image/webp' ) ) {
+			$this->assertSame( 'image/webp', $saved['mime-type'] );
+			$this->assertSame( 'canola-100x75.webp', $saved['file'] );
```

Comment:
Shouldn't this result in `canola-100x75-jpg.webp`? Why is that extra original extension suffix missing here? Is that a potential problem we need to fix?

Reply:
I looked into this a little. I think it is because of the way the test is structured with `generate_filename` called with no arguments.  Our code in this function relies on `$extension` being passed, which is what happens when core generates subsized images. I'll see if we can improve this and also test the core calls to create the `-scaled` and `-rotated` names which may not be adding the additional suffix, but probably should.

Reply:
Turns out this was happening because of the way the test was structured. By first generating the filename, then passing that to save the code was bypassing the for altered extension - that only happens when you don't pass a filename to save, see https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/class-wp-image-editor-imagick.php#L717-L719 

Fixed this in c0367e179c6a138e8c56d9c47a66dae659874f3c by altering the test slightly, calling save without first generating the filename. Essentially if you pass a filename to save, it uses that filename, that is a requirement for being able to overwrite files.